Assets/Scripts/Interactables/CocktailMiniGame.cs

```diff
@@ -0,0 +1,374 @@
+using FMS.TapperRedone.Characters;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI; // For UI elements
+
+using FMS.TapperRedone.Data;
+using FMS.TapperRedone.Inventory;
+using FMS.TapperRedone.Managers;
+
+namespace FMS.TapperRedone.Interactables
+{
+
+    public class CocktailMiniGame : InteractableObject
+    {
+
+        // References to the sprites for the arrows
+        public Sprite leftUIPressed;
+        public Sprite leftUIUnpressed;
+        public Sprite rightUIPressed;
+        public Sprite rightUIUnpressed;
+
+        public SpriteRenderer cocktailShakerRenderer; // Assign this with the Cocktail Shaker's Sprite Renderer in the Inspector
+        public GameObject leftArrow; // Assign the LeftArrow object here
+        public GameObject rightArrow; // Assign the RightArrow object here
+
+        private PlayerInteraction player; // Store player reference
+        private bool detectedInputDuringMiniGame = false;
+        public Item itemToSpawn;
+
+        public KeyCode leftKey = KeyCode.A;  // Key for left arrow
+        public KeyCode rightKey = KeyCode.D; // Key for right arrow
+        public KeyCode Quit = KeyCode.Q; //hit Q to quit, escape pauses game entirely
+
+        public float arrowSwitchSpeed = 0.5f; // Speed at which arrows switch, adjust for difficulty
+        private bool leftArrowActive = true;  // Determine which arrow is currently active
+        private bool isMiniGameActive = false; // State of the minigame
+
+        private float scoreMultiplier = 1f;  // Score multiplier based on timing
+        private float perfectHitWindow = 0.2f;  // Time window for a perfect hit
+
+        public CanvasGroup cocktailCanvasGroup; // Reference to your Canvas Group for the mini-game UI
+
+        [Tooltip("Duration of the mini-game in seconds.")]
+        public float miniGameDuration = 10f; // Default to 10 seconds, can be adjusted in the inspector
+
+        private Coroutine rhythmCoroutine; // Coroutine to manage rhythm game
+
+
+        // Start is called before the first frame update
+        void Start()
+        {
+            GameManager.OnGameStateChanged += OnGameStateChanged;
+
+            // Set the Cocktail Shaker to be visible at the start
+            if (cocktailShakerRenderer != null)
+            {
+                cocktailShakerRenderer.enabled = true;
+            }
+
+            // Ensure arrows are hidden until the minigame starts
+            if (leftArrow != null) leftArrow.SetActive(false);
```

Comment:
If you do the non-null check at Awake instead of at Start (I think?), you could also do a Debug.LogWarning saying "hey dont forget  to assign this variable"